File: 122.maxProfit.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() <= 1) return 0;
        int slow = 0;
        int fast = 1;
        int profit = 0;

        while(fast < prices.size()) {
            if (prices[slow] < prices[fast]) {
                profit += prices[fast] - prices[slow];
            }
            fast++;
            slow++;
        }

        return profit;
    }
};

File: 122.maxProfit.go
func maxProfit(prices []int) int {
    if len(prices) <= 1 {
        return 0;
    }

    pre := prices[0]
    ans := 0

    for _, price := range(prices) {
        if price - pre > 0 {
            ans += price - pre
        }
        pre = price
    }

    return ans
}